File: test_data/code/python/1221-split-a-string-in-balanced-strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        res = 0
        count = 0
        for c in s:
            if c == 'L':
                count += 1
            else:
                count -= 1
            if count == 0:
                res += 1
        return res

File: test_data/code/java/1221-split-a-string-in-balanced-strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int count = 0;
        int balance = 0;
        
        for (char c : s.toCharArray()) {
            balance += (c == 'L') ? -1 : 1;
            if (balance == 0) {
                count++;
            }
        }
        
        return count;
    }
}